docs/content/preview/api/ysql/the-sql-language/statements/dml_insert.md

```diff
@@ -78,12 +78,42 @@ the INSERT ON CONFLICT variant is often colloquially referred to as "upsert".
 
 ### *returning_clause*
 
+The `RETURNING` clause causes the `INSERT` statement to compute and return value(s) based on each row actually inserted (or updated, if an `ON CONFLICT DO UPDATE` clause is used). 
+
+This is useful for obtaining values that were supplied by defaults, such as a serial sequence number. Any expression using the table's columns is allowed. 
+
+`RETURNING` column list is identical to that of the output list of `SELECT`. 
+
+Only successfully inserted or updated rows are returned.
+
+
 ### *column_values*
 
 ### *conflict_target*
 
+Specifies which conflicts ON CONFLICT takes the alternative action on by choosing arbiter indexes. 
+Either performs unique index inference, or names a constraint explicitly. 
+
+For ON CONFLICT DO NOTHING, it is optional to specify a conflict_target; when omitted, conflicts with all usable constraints (and unique indexes) are handled. 
+
+For ON CONFLICT DO UPDATE, a conflict_target must be provided.
+
+
+Specify which arbiter index to use to match rows for `ON CONFLICT`.
+`ON CONFLICT` either checks a unique index, or specifies a constraint explicitly.
+When using `ON CONFLICT DO UPDATE`, the `conflict_target` is required.
+When using `ON CONFLICT DO NOTHING`, the `conflict_target` is optional. If omitted, it will conflict using all usable constraints and unique indexes of the table.
+
 ### *conflict_action*
 
+`conflict_action` clause specifies an action that happens when you use `ON CONFLICT`. The action includes `DO NOTHING`, or `DO UPDATE`. `DO UPDATE` clause specifies the `UPDATE` action to be executed on conflicted rows.
```

Comment:
nit: 
> happens when you use `ON CONFLICT`
happens _instead of_ a unique constraint violation if that is hit -- not all the time. Maybe this can be clarified.

Reply:
Did in https://github.com/yugabyte/yugabyte-db/pull/20553/commits/aba1f30335ac22b1afcc9efbb93660993f738d6a